src/monattack.cpp

```diff
@@ -131,6 +137,9 @@ void mattack::howl(monster *z, int index)
 
 void mattack::rattle(monster *z, int index)
 {
+    if( z->friendly ) {
+        return; // TODO: handle friendly monsters
```

Comment:
Alert if the rattler is threatened: hostile critter in proximity, just like you the player might have been.